test/spec/BuildSpec.hs

```diff
@@ -42,6 +42,42 @@ spec = do
           stderr output `shouldContain` "Couldn't match type"
           exitCode output `shouldBe` ExitFailure 1
 
+        describe "addPackage" $ do
+          it "allows adding packages to projects within the project's environment" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                const env = garn.mkEnvironment(undefined, garn.nix.nixStrLit`
+                  mkdir dist
+                  echo build-content > dist/build-artifact
+                `);
+                export const project = garn.mkProject({
+                  description: "",
+                  defaultEnvironment: env,
+                }, {})
+                  .addPackage("build", "mv dist/build-artifact $out/build-artifact");
+              |]
+            output <- runGarn ["build", "project"] "" repoDir Nothing
+            onTestFailureLog output
+            readFile "result/build-artifact" `shouldReturn` "build-content\n"
+            exitCode output `shouldBe` ExitSuccess
+
+          it "allows adding packages to projects without a default environment" $ \onTestFailureLog -> do
```

Comment:
```suggestion
          it "allows adding packages to projects" $ \onTestFailureLog -> do
```

And make this the first test case of the two added ones? This feels like the more basic test.

Reply:
This test case was deleted since we don't support `addPackage` on non-default environments